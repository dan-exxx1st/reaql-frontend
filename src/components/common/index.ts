export { default as DialogHeader } from './DialogHeader';
export { default as ContactItem } from './ContactsItem';
export { default as UserHeader } from './UserHeader';
export { default as DialogTextField } from './DialogTextField';
export { default as MessageItem } from './MessageItem';
export { default as DialogHeaderBackdrop } from './DialogHeaderBackdrop';
export { default as MessageSearch } from './MessageSearch';
