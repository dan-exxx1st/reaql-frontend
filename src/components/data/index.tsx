export { default as MessageListWithData } from './MessageListWithData';
export { default as DialogHeaderWithData } from './DialogHeaderWithData';
