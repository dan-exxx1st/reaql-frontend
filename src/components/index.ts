export { default as DialogListItem } from './common/DialogListItem';
export { default as EmojiPicker } from './common/EmojiPicker';
