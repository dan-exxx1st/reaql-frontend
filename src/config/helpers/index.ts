import { useState, useEffect } from 'react';

export function useLoaded({ src }: { src?: HTMLImageElement['src'] }) {
	const [loaded, setLoaded] = useState('false');

	useEffect(() => {
		if (!src) {
			return undefined;
		}

		setLoaded('loading');

		let active = true;
		const image = new Image();
		image.src = src;
		image.onload = () => {
			if (!active) {
				return;
			}
			setLoaded('loaded');
		};
		image.onerror = () => {
			if (!active) {
				return;
			}
			setLoaded('error');
		};

		return () => {
			active = false;
		};
	}, [src]);

	return loaded;
}
