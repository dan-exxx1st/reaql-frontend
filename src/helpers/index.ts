import { parse } from 'querystring';

export function getDialogIdFromSearch(search: string) {
    const params = parse(search);

    const dialogId = Object.entries(params)
        .map(([key, value]) => key.indexOf('dialog') !== -1 && value)
        .toString();

    return dialogId;
}
