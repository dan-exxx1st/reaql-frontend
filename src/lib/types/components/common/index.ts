import Theme from 'helpers/styled';
import { Dialog, Maybe, Message, Message_Statuses } from 'lib/graphql/types';
import { Dispatch, SetStateAction } from 'react';

export interface IUserHeaderProps {
    className?: string;
    setSearchUserOpened?: (val: boolean) => void;
}

export interface IMessageSearchProps {
    onClick?: () => void;
    value?: string;
    messageFilter?: (value: string) => void | Dispatch<SetStateAction<string>>;
}

export interface IMessageItemProps extends Message {
    className?: string;
}

export interface IDialogsItemProps {
    id: string;
    name: string;
    surname: string;
    avatar: string;
    unreadMessages: number | null | undefined;
    lastMessageStatus: Message_Statuses | null | undefined;
    lastMessage: string | null | undefined;
    lastMessageDate: string | null | undefined;
}

export interface IMessageListProps {
    data?: {
        messages: Message[];
    };
    loading?: boolean;
    subscribeToNewMessages?: () => void;
    className?: string;
    filterValue?: string;
}

export interface IDialogListProps {
    dialogFilter: string;
    dialogs?: Maybe<Dialog[]>;
    className?: string;
}

export interface IDialogTextFieldProps {
    className?: string;
}

export interface IDialogHeaderProps {
    className?: string;
    messageFilter?: Dispatch<SetStateAction<string>>;
    filterValue?: string;
    name: string;
    onlineStatus?: string | null;
    avatar?: string;
    loading?: boolean;
}

export interface ISignUpFormProps {
    email: string;
    name: string;
    surname: string;
    password: string;
    changeFormValues: (field: string, value: string) => void;
    handleSubmit: () => void;
    theme?: typeof Theme;
}

export interface ISignInFormProps {
    _handleSignIn: (payload: {
        email: string;
        password: string;
        rememberMe: boolean;
    }) => void;
}

export interface ISideBarProps {
    data?: {
        dialogs: Dialog[];
    };
    loading: boolean;
    subscribeToNewDialogs?: () => void;
    setSearchUserOpened: (val: boolean) => void;
}
