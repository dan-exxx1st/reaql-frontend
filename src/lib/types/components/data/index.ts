export interface ISideBarWithDataProps {
    setSearchUserOpened?: (val: boolean) => void;
}
