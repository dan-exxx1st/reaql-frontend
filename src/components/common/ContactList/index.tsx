import React, { FC } from 'react';

import { IContactListProps } from 'lib/types/components/common';

import { StyledContactListWrapper, StyledContactListItem } from './style';

const ContactList: FC<IContactListProps> = ({ contacts, ...otherProps }) => {
    return (
        <StyledContactListWrapper {...otherProps}>
            {contacts &&
                contacts.map((contact) => (
                    <StyledContactListItem key={contact.id} {...contact} />
                ))}
        </StyledContactListWrapper>
    );
};

export default ContactList;
