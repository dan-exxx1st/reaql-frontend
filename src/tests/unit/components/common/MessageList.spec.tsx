import React from 'react';
import { MessageList } from 'components/common';
import { mountWithApolloAndStyled } from 'tests/helpers/withApollo';

import { MessageListData } from 'tests/__mocks__/data/unit';

describe('<MessageList />', () => {
    describe('Snapshots', () => {
        it('Should render correctly', () => {
            const wrapper = mountWithApolloAndStyled(
                <MessageList messages={MessageListData} />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });
});
