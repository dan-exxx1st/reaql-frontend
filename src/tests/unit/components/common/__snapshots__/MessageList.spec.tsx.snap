// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageList /> Snapshots Should render correctly 1`] = `
.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 40px;
  height: 40px;
}

.c8 {
  font-family: 'Roboto',sans-serif;
  font-weight: normal;
  font-size: 16px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 19px;
  color: #000000;
  margin: 0;
}

.c10 {
  font-family: 'Roboto',sans-serif;
  font-weight: normal;
  font-size: 12px;
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
  line-height: 14px;
  color: #000000;
  margin: 0;
}

.c12 {
  font-family: 'Roboto',sans-serif;
  font-weight: normal;
  font-size: 14px;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  color: #000000;
  margin: 0;
}

.c0 {
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  width: auto;
  height: auto;
}

.c3 {
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 0;
  margin: 0;
}

.c5 {
  margin-right: 15px;
}

.c9 {
  margin-right: 10px;
  color: #252525;
}

.c11 {
  color: #A5A5A5;
}

.c13 {
  color: #000000;
}

.c6 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 40px;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
}

.c2 {
  margin-bottom: 20px;
}

<div
  className="c0 c1"
>
  <div
    className="c0 c2"
  >
    <img
      alt="Icon component"
      className="c3 c4 c5"
      src="Person.svg"
    />
    <div
      className="c0 c6"
    >
      <div
        className="c0 c7"
      >
        <p
          className="c8 c9"
        >
          Test User
        </p>
        <p
          className="c10 c11"
        >
          12:00 am
        </p>
      </div>
      <p
        className="c12 c13"
      >
        Test message
      </p>
    </div>
  </div>
  <div
    className="c0 c2"
  >
    <img
      alt="Icon component"
      className="c3 c4 c5"
      src="Person.svg"
    />
    <div
      className="c0 c6"
    >
      <div
        className="c0 c7"
      >
        <p
          className="c8 c9"
        >
          Test User 2
        </p>
        <p
          className="c10 c11"
        >
          12:00 am
        </p>
      </div>
      <p
        className="c12 c13"
      >
        Test message 2
      </p>
    </div>
  </div>
</div>
`;
